rskj-core/src/test/java/co/rsk/peg/BridgeSupportTestIntegration.java

```diff
@@ -4034,121 +4239,217 @@ private BridgeSupport getBridgeSupportWithMocksForWhitelistTests(LockWhitelist m
+            .withBtcBlockStoreFactory(btcBlockStoreFactory)
+            .withExecutionBlock(executionBlock)
+            .withFederationSupport(federationSupport)
+            .build();
     }
 
     private BridgeSupport getBridgeSupport(
-            BridgeConstants constants,
-            BridgeStorageProvider provider,
-            Repository track,
-            BridgeEventLogger eventLogger,
-            BtcLockSenderProvider btcLockSenderProvider,
-            Block executionBlock,
-            BtcBlockStoreWithCache.Factory blockStoreFactory) {
-
-        return getBridgeSupport(
-                constants,
-                provider,
-                track,
-                eventLogger,
-                btcLockSenderProvider,
-                executionBlock,
-                blockStoreFactory,
-                mock(ActivationConfig.ForBlock.class)
-        );
+        BridgeStorageProvider provider,
+        Repository track,
+        BtcBlockStoreWithCache.Factory blockStoreFactory,
+        ActivationConfig.ForBlock activations) {
+
+        return bridgeSupportBuilder
+            .withBridgeConstants(bridgeConstants)
+            .withProvider(provider)
+            .withRepository(track)
+            .withBtcBlockStoreFactory(blockStoreFactory)
+            .withActivations(activations)
+            .build();
     }
 
     private BridgeSupport getBridgeSupport(
-            BridgeConstants constants,
-            BridgeStorageProvider provider,
-            Repository track,
-            BridgeEventLogger eventLogger,
-            BtcLockSenderProvider btcLockSenderProvider,
-            Block executionBlock,
-            BtcBlockStoreWithCache.Factory blockStoreFactory,
-            ActivationConfig.ForBlock activations) {
-
-        if (blockStoreFactory == null) {
-            blockStoreFactory = mock(BtcBlockStoreWithCache.Factory.class);
-        }
+        BridgeStorageProvider provider,
+        Repository track,
+        BtcBlockStoreWithCache.Factory blockStoreFactory,
+        FederationSupport federationSupport) {
 
-        FederationSupport federationSupport = mock(FederationSupport.class);
-        return new BridgeSupport(
-            constants,
-            provider,
-            eventLogger,
-            btcLockSenderProvider,
-            new PeginInstructionsProvider(),
-            track,
-            executionBlock,
-            new Context(constants.getBtcParams()),
-            federationSupport,
-            feePerKbSupport,
-            blockStoreFactory,
-            activations,
-            signatureCache
-        );
+        return bridgeSupportBuilder
+            .withBridgeConstants(bridgeConstants)
+            .withProvider(provider)
+            .withRepository(track)
+            .withBtcBlockStoreFactory(blockStoreFactory)
+            .withFederationSupport(federationSupport)
+            .build();
+    }
+
+    private BridgeSupport getBridgeSupport(
```

Comment:
Can we avoid adding all these new methods and use BridgeSupportBuilder instead? That was the original purpose

Reply:
done